### PR TITLE
#101 Copy content from another course bug

### PR DESCRIPTION
**Pull request by goodviber**
### Context
There is a bug with the copy content from another course feature which caused [this](https://sentry.io/organizations/dfe-teacher-services/issues/3300113124/?environment=production&project=1377944&query=is%3Aunresolved+url%3A%22https%3A%2F%2Fwww.publish-teacher-training-courses.service.gov.uk%2Fpublish%2Forganisations%2F13J%2F2022%2Fcourses%2FR030%2Fabout%22+url%3A%22https%3A%2F%2Fwww.publish-teacher-training-courses.service.gov.uk%2Fpublish%2Forganisations%2F13J%2F2022%2Fcourses%2FR030%2Fabout%22) bug in production.

You can also copy content from the course you are trying to copy to, which shouldn't be the case.

Steps to reproduce
Create a brand new course
Click on the course
Click on Description
Click the Change link adjacent to About this course
You are able to select the course that you are currently copying to.
If you select the course and click Copy content you should get the error attached to this card.
Tech notes
source_course.enrichments is an empty array
Success
We have a test that replicates this bug
We have a fix for it
We mark the Sentry error as resolved

### Changes proposed in this pull request
Add some test coverage to the copy list options and fix `fetch_course_list_to_copy_from`
### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

File: app/controllers/concerns/copy_course_content.rb [hunk 1]
```diff
@@ -32,7 +32,7 @@ def fetch_course_list_to_copy_from
     @courses_by_accrediting_provider = ::Courses::Fetch.by_accrediting_provider(@provider)
     @self_accredited_courses = @courses_by_accrediting_provider.delete(@provider.provider_name)
 
-    @courses_by_accrediting_provider = @courses_by_accrediting_provider.reject { |c| c == course.id }
+    @courses_by_accrediting_provider&.transform_values! { |v| v.reject { |x| x.id == course.id } }
```
Comment (avinhurry) [inline app/controllers/concerns/copy_course_content.rb:35]: 👍🏼

File: spec/features/publish/courses/editing_course_information_spec.rb [hunk 2]
```diff
@@ -51,6 +51,18 @@
...
+        provider_code: provider.provider_code, recruitment_cycle_year: provider.recruitment_cycle_year, course_code: course.course_code,
+      )
+
+      list_options = publish_course_information_page.copy_content.copy_options
+      expect(Course.count).to eq 3
+      expect(list_options.size).to eq 3
+      expect(list_options.shift).to eq("Pick a course")
+      expect(list_options.any? { |x| x[@course.name] }).to be_falsey
+    end
+
```
Comment (avinhurry) [inline spec/features/publish/courses/editing_course_information_spec.rb:65]: This test passes on master. 

Can we maybe have a test that lists out all the courses and lists out all the course options in the list, demonstrating that one of the courses is not in the list?

**Review by avinhurry — COMMENTED** [review]
Tested locally, implementation looks good! 

I had a comment regarding the test inline.

File: spec/features/publish/courses/editing_course_information_spec.rb (hunk 2 above)
Comment (goodviber) [reply spec/features/publish/courses/editing_course_information_spec.rb:65]: I think this is what it does, there are 3 courses, and 3 text options, the first one being 'pick a course', the remaining 2 being course2 and course3. The course name is not in the list `expect(list_options.any? { |x| x[@course.name] }).to be_falsey` , or did you mean something else?

File: docs/support_playbook.md [hunk 3]
```diff
@@ -62,7 +62,7 @@ p.courses.update(accredited_body_code: "1YK")
 When doing this, you should check if the user already exists in prod. When testing locally, you'll likely be working with a sanitised database dump.
 
 ```ruby
-user = User.find(email: "jon@email.com")
```
Comment (goodviber) [inline docs/support_playbook.md:65]: not really part of the ticket but couldn't resist

**Pull request merged**